src/librustdoc/test.rs

```diff
@@ -273,13 +273,28 @@ fn runtest(test: &str, cratename: &str, cfgs: Vec<String>, libs: SearchPaths,
                     } else if count == 0 && compile_fail == true {
                         panic!("test compiled while it wasn't supposed to")
                     }
+                    if count > 0 && error_codes.len() > 0 {
+                        let out = String::from_utf8(data.lock().unwrap().to_vec()).unwrap();
```

Comment:
We do, though these errors tend to get emitted and forgotten.  Unless you can think of a way I'm not thinking of?

I'm working on a refactor to move errors into their own crate and to be able to have multiple emits (default, expanded, json, etc).  May make sense to let you buffer them up and test them, not sure.